application/org.openjdk.jmc.ui.common/src/main/java/org/openjdk/jmc/ui/common/util/ThemeUtils.java

```diff
@@ -1,22 +1,46 @@
 package org.openjdk.jmc.ui.common.util;
 
+import org.eclipse.jface.util.PropertyChangeEvent;
+
 import org.eclipse.jface.preference.JFacePreferences;
 import org.eclipse.jface.resource.ColorRegistry;
+import org.eclipse.jface.util.IPropertyChangeListener;
 import org.eclipse.swt.graphics.RGB;
 import org.eclipse.ui.PlatformUI;
 
 public class ThemeUtils {
+
+	private static boolean isCurrentThemeDark = isDarkMode();
```

Comment:
**suggestion:** 

> But seems like spotbugs is not happy with updating static variable isCurrentThemeDark from the listener

I don't know if it's alright by JMC spotbug rules to write this global state like this

```suggestion
	private static AtomicReference<Boolean> isCurrentThemeDark = new AtomicReference<>(isDarkMode());
```

and then access it with `AtomicReference::get` / `AtomicReference::set`.

Reply:
Thanks for the suggestion. I have tried fixing it with another simple approach and updated the PR with that change. Spotbugs is also happy with that. Can you please have a look? If it doesn't look good, I will try this AtomicReference approach. Thanks in advance.

Reply:
@Suchitainf Your alternative approach is perfectly fine :)